src/application_settings/type_notation_helper.py

```diff
@@ -1,26 +1,18 @@
 # pylint: disable=consider-alternative-union-syntax, useless-suppression
 """Defines type aliases that handle notational differences between python versions."""
 import sys
-from collections.abc import Callable
 from pathlib import Path
 from types import ModuleType
-from typing import Any
 
 if sys.version_info >= (3, 10):
     from typing import TypeAlias
 
     PathOrStr: TypeAlias = Path | str
-    PathOpt: TypeAlias = Path | None
-    LoaderOpt: TypeAlias = Callable[[Path], dict[str, Any]] | None
-    SaverOpt: TypeAlias = Callable[[Path, dict[str, Any]], None] | None
     ModuleTypeOpt: TypeAlias = ModuleType | None
 else:
     from typing import Union
 
     from typing_extensions import TypeAlias
 
     PathOrStr: TypeAlias = Union[Path, str]
-    PathOpt: TypeAlias = Union[Path, None]
-    LoaderOpt: TypeAlias = Union[Callable[[Path], dict[str, Any]], None]
-    SaverOpt: TypeAlias = Union[Callable[[Path, dict[str, Any]], None], None]
     ModuleTypeOpt: TypeAlias = Union[ModuleType, None]
```

Comment:
Do we still need this one, or should we also replace it with `Optional[ModuleType]`. I'm also not really sure anymore why we did it this way initially?

Reply:
I agree, replaced with `Optional[ModuleType]`.
We started doing this because `|` is not supported by python 3.9 and therefore we used `Union`, without realizing that for `| None` we could better use `Optional`.
I think that next year we can drop 3.9 and remove these typevars alltogether